### ISSUE TITLE
[Bug]: Integration not loading after update to 2023.9.3

### ISSUE THREAD

**Issue by boku78** [wlcrs/huawei_solar#487]
### Describe the issue

Hello,

i just updated homeassitant to 2023.9.3 and after the update the integration doesn't load anymore.

Here is the error message:

Logger: homeassistant.setup
Source: loader.py:826
First occurred: 07:25:28 (1 occurrences)
Last logged: 07:25:28
Setup failed for custom integration huawei_solar: Unable to import component: Exception importing custom_components.huawei_solar

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/loader.py", line 818, in get_component
    ComponentProtocol, importlib.import_module(self.pkg_path)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1147, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/config/custom_components/huawei_solar/__init__.py", line 25, in <module>
    from huawei_solar import HuaweiSolarBridge, HuaweiSolarException, InvalidCredentials
  File "/usr/local/lib/python3.11/site-packages/huawei_solar/__init__.py", line 6, in <module>
    from .bridge import HuaweiSolarBridge  # noqa
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/huawei_solar/bridge.py", line 16, in <module>
    from .huawei_solar import DEFAULT_BAUDRATE, DEFAULT_SLAVE, DEFAULT_TCP_PORT, AsyncHuaweiSolar, Result
  File "/usr/local/lib/python3.11/site-packages/huawei_solar/huawei_solar.py", line 17, in <module>
    from pymodbus.payload import BinaryPayloadBuilder, BinaryPayloadDecoder
  File "/usr/local/lib/python3.11/site-packages/pymodbus/payload.py", line 28, in <module>
    class BinaryPayloadBuilder:
  File "/usr/local/lib/python3.11/site-packages/pymodbus/payload.py", line 42, in BinaryPayloadBuilder
    self, payload=None, byteorder=Endian.Little, wordorder=Endian.Big, repack=False
                                  ^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/enum.py", line 784, in __getattr__
    raise AttributeError(name) from None
AttributeError: Little

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/setup.py", line 215, in _async_setup_component
    component = integration.get_component()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 826, in get_component
    raise ImportError(f"Exception importing {self.pkg_path}") from err
ImportError: Exception importing custom_components.huawei_solar

Kind regards
Boris

### Bescribe your Huawei Solar Setup

Inverter Type:
Inverter Firmware version:
SDongle present: yes / no
Power meter present: three phase / single phase / no
Battery: LG RESU xxkWh / LUNA2000 xxkWh / None
Battery Firmware version:

### How do you connect to the inverter?

Via the `SUN2000-<serial_no>` WiFi

### Upload your Diagnostics File

There is no button "Download diagnostics":

![grafik](https://github.com/wlcrs/huawei_solar/assets/13376487/194c1a9c-f3c8-4714-917e-6ce6c9c3dc00)

### Upload your relevant debug logs

```text
...
2023-09-27 08:44:20.489 DEBUG (MainThread) [homeassistant.components.tplink.coordinator] Finished fetching 192.168.2.69 data in 0.155 seconds (success: True)
2023-09-27 08:44:20.490 DEBUG (MainThread) [homeassistant.core] Bus:Handling <Event state_changed[L]: entity_id=sensor.tv_strom_voltage, old_state=<state sensor.tv_strom_voltage=233.6; state_class=measurement, unit_of_measurement=V, device_class=voltage, friendly_name=TV Strom Spannung @ 2023-09-27T08:44:10.493726+02:00>, new_state=<state sensor.tv_strom_voltage=233.7; state_class=measurement, unit_of_measurement=V, device_class=voltage, friendly_name=TV Strom Spannung @ 2023-09-27T08:44:20.490400+02:00>>
2023-09-27 08:44:20.491 DEBUG (MainThread) [homeassistant.components.websocket_api.http.connection] [547426789184] Boris from 192.168.2.21 (Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/118.0): Sending {"id":3,"type":"event","event":{"c":{"sensor.tv_strom_voltage":{"+":{"s":"233.7","lc":1695797060.4904,"c":"01HBAP38WA4N8HPYMZP2EHHMZG"}}}}}
2023-09-27 08:44:20.494 DEBUG (Recorder) [homeassistant.components.recorder.core] Processing task: EventTask(event=<Event state_changed[L]: entity_id=sensor.tv_strom_voltage, old_state=<state sensor.tv_strom_voltage=233.6; state_class=measurement, unit_of_measurement=V, device_class=voltage, friendly_name=TV Strom Spannung @ 2023-09-27T08:44:10.493726+02:00>, new_state=<state sensor.tv_strom_voltage=233.7; state_class=measurement, unit_of_measurement=V, device_class=voltage, friendly_name=TV Strom Spannung @ 2023-09-27T08:44:20.490400+02:00>>)
2023-09-27 08:44:20.502 DEBUG (MainThread) [homeassistant.components.fritz.switch] Specific WiFiNetwork response: GetInfo={'NewEnable': True, 'NewStatus': 'Up', 'NewMaxBitRate': 'Auto', 'NewChannel': 6, 'NewSSID': 'BoNaK', 'NewBeaconType': '11i', 'NewX_AVM-DE_PossibleBeaconTypes': 'None,11i,WPAand11i,11iandWPA3', 'NewMACAddressControlEnabled': False, 'NewStandard': 'n', 'NewBSSID': '44:4E:6D:20:C1:12', 'NewBasicEncryptionModes': 'None', 'NewBasicAuthenticationMode': 'None', 'NewMaxCharsSSID': 32, 'NewMinCharsSSID': 1, 'NewAllowedCharsSSID': '0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz !"#$%&\'()*+,-./:;<=>?@[\\]^_`{|}~', 'NewMinCharsPSK': 64, 'NewMaxCharsPSK': 64, 'NewAllowedCharsPSK': '0123456789ABCDEFabcdef', 'NewX_AVM-DE_FrequencyBand': '2400', 'NewX_AVM-DE_WLANGlobalEnable': True}
2023-09-27 08:44:20.502 DEBUG (MainThread) [homeassistant.components.fritz.switch] Specific PortForward response: {'NewRemoteHost': '0.0.0.0', 'NewExternalPort': 443, 'NewProtocol': 'TCP', 'NewInternalPort': 443, 'NewInternalClient': '192.168.2.90', 'NewEnabled': True, 'NewPortMappingDescription': 'HTTPS-Server', 'NewLeaseDuration': 0}
2023-09-27 08:44:20.502 DEBUG (MainThread) [homeassistant.components.fritz.common] Checking devices for FRITZ!Box device 192.168.2.1
2023-09-27 08:44:20.555 DEBUG (MainThread) [zeroconf] IPv6 scope_id 0 associated to the receiving interface
2023-09-27 08:44:20.555 DEBUG (MainThread) [zeroconf] Received from '::ffff:192.168.2.21':5353 [socket 12 (('::', 5353, 0, 0))]: <DNSIncoming:{id=0, flags=0, truncated=False, n_q=1, n_ans=0, n_auth=0, n_add=0, questions=[a[question,QM,in,BRN30055C2C6A1E.local.]], answers=[]}> (39 bytes) as [b'\x00\x00\x00\x00\x00\x01\x00\x00\x00\x00\x00\x00\x0fBRN30055C2C6A1E\x05local\x00\x00\x01\x00\x01']
2023-09-27 08:44:20.560 DEBUG (MainThread) [zeroconf] IPv6 scope_id 0 associated to the receiving interface
2023-09-27 08:44:20.561 DEBUG (MainThread) [zeroconf] Received from '::ffff:192.168.2.38':5353 [socket 12 (('::', 5353, 0, 0))]: <DNSIncoming:{id=0, flags=33792, truncated=False, n_q=0, n_ans=1, n_auth=0, n_add=0, questions=[], answers=[record[a,in-unique,BRN30055C2C6A1E.local.]=240.0/239,192.168.2.38]}> (49 bytes) as [b'\x00\x00\x84\x00\x00\x00\x00\x01\x00\x00\x00\x00\x0fBRN30055C2C6A1E\x05local\x00\x00\x01\x80\x01\x00\x00\x00\xf0\x00\x04\xc0\xa8\x02&']
2023-09-27 08:44:20.569 DEBUG (MainThread) [zeroconf] IPv6 scope_id 0 associated to the receiving interface
2023-09-27 08:44:20.570 DEBUG (MainThread) [zeroconf] Received from '::ffff:192.168.2.21':5353 [socket 12 (('::', 5353, 0, 0))]: <DNSIncoming:{id=0, flags=0, truncated=False, n_q=1, n_ans=0, n_auth=0, n_add=0, questions=[a[question,QM,in,BRN30055C2C6A1E.local.]], answers=[]}> (39 bytes) as [b'\x00\x00\x00\x00\x00\x01\x00\x00\x00\x00\x00\x00\x0fBRN30055C2C6A1E\x05local\x00\x00\x01\x00\x01']
2023-09-27 08:44:20.651 DEBUG (MainThread) [zeroconf] IPv6 scope_id 0 associated to the receiving interface
2023-09-27 08:44:20.652 DEBUG (MainThread) [zeroconf] Received from '::ffff:192.168.2.62':5353 [socket 12 (('::', 5353, 0, 0))]: <DNSIncoming:{id=0, flags=33792, truncated=False, n_q=0, n_ans=1, n_auth=0, n_add=0, questions=[], answers=[record[a,in-unique,192-168-2-62.local.]=3599.0/3599,192.168.2.62]}> (46 bytes) as [b'\x00\x00\x84\x00\x00\x00\x00\x01\x00\x00\x00\x00\x0c192-168-2-62\x05local\x00\x00\x01\x80\x01\x00\x00\x0e\x0f\x00\x04\xc0\xa8\x02>']
2023-09-27 08:44:20.660 DEBUG (MainThread) [zeroconf] IPv6 scope_id 0 associated to the receiving interface
2023-09-27 08:44:20.661 DEBUG (MainThread) [zeroconf] Received from '::ffff:192.168.2.62':5353 [socket 12 (('::', 5353, 0, 0))]: <DNSIncoming:{id=0, flags=33792, truncated=False, n_q=0, n_ans=1, n_auth=0, n_add=0, questions=[], answers=[record[a,in-unique,192-168-2-62.local.]=3600.0/3600,192.168.2.62]}> (46 bytes) as [b'\x00\x00\x84\x00\x00\x00\x00\x01\x00\x00\x00\x00\x0c192-168-2-62\x05local\x00\x00\x01\x80\x01\x00\x00\x0e\x10\x00\x04\xc0\xa8\x02>']
2023-09-27 08:44:20.853 DEBUG (Thread-2 (_thread_main)) [paho.mqtt.client] Received PUBLISH (d0, q0, r0, m0), 'zigbee2mqtt/Wohnzimmer TV Strom', ...  (246 bytes)
2023-09-27 08:44:20.855 DEBUG (MainThread) [homeassistant.components.mqtt.client] Received message on zigbee2mqtt/Wohnzimmer TV Strom (qos=0): b'{"child_lock":"LOCK","current":0.24,"energy":55.91,"indicator_mode":null,"linkquality":69,"power":13,"power_outage_memory":"on","state":"ON","update":{"installed_version":-1,"latest_version":-1,"state":null},"update_available":null,"voltage":233}'
2023-09-27 08:44:20.856 DEBUG (MainThread) [homeassistant.components.mqtt.models] Rendering incoming payload '{"child_lock":"LOCK","current":0.24,"energy":55.91,"indicator_mode":null,"linkquality":69,"power":13,"power_outage_memory":"on","state":"ON","update":{"installed_version":-1,"latest_version":-1,"state":null},"update_available":null,"voltage":233}' with variables {'entity_id': 'sensor.wohnzimmer_tv_strom_power', 'name': 'Leistung', 'this': <template TemplateStateFromEntityId(sensor.wohnzimmer_tv_strom_power)>} with default value 'default' and Template<template=({{ value_json.power }}) renders=8>
2023-09-27 08:44:20.856 DEBUG (MainThread) [homeassistant.components.mqtt.models] Rendering incoming payload '{"child_lock":"LOCK","current":0.24,"energy":55.91,"indicator_mode":null,"linkquality":69,"power":13,"power_outage_memory":"on","state":"ON","update":{"installed_version":-1,"latest_version":-1,"state":null},"update_available":null,"voltage":233}' with variables {'entity_id': 'sensor.wohnzimmer_tv_strom_energy', 'name': 'Energie', 'this': <template TemplateStateFromEntityId(sensor.wohnzimmer_tv_strom_energy)>} with default value 'default' and Template<template=({{ value_json.energy }}) renders=8>
2023-09-27 08:44:20.857 DEBUG (MainThread) [homeassistant.components.mqtt.models] Rendering incoming payload '{"child_lock":"LOCK","current":0.24,"energy":55.91,"indicator_mode":null,"linkquality":69,"power":13,"power_outage_memory":"on","state":"ON","update":{"installed_version":-1,"latest_version":-1,"state":null},"update_available":null,"voltage":233}' with variables {'entity_id': 'switch.wohnzimmer_tv_strom', 'name': None, 'this': <template TemplateStateFromEntityId(switch.wohnzimmer_tv_strom)>} and Template<template=({{ value_json.state }}) renders=8>
2023-09-27 08:44:20.857 DEBUG (MainThread) [homeassistant.components.mqtt.models] Rendering incoming payload '{"child_lock":"LOCK","current":0.24,"energy":55.91,"indicator_mode":null,"linkquality":69,"power":13,"power_outage_memory":"on","state":"ON","update":{"installed_version":-1,"latest_version":-1,"state":null},"update_available":null,"voltage":233}' with variables {'entity_id': 'select.wohnzimmer_tv_strom_power_outage_memory', 'name': 'Power outage memory', 'this': <template TemplateStateFromEntityId(select.wohnzimmer_tv_strom_power_outage_memory)>} and Template<template=({{ value_json.power_outage_memory }}) renders=8>
2023-09-27 08:44:20.858 DEBUG (MainThread) [homeassistant.components.mqtt.models] Rendering incoming payload '{"child_lock":"LOCK","current":0.24,"energy":55.91,"indicator_mode":null,"linkquality":69,"power":13,"power_outage_memory":"on","state":"ON","update":{"installed_version":-1,"latest_version":-1,"state":null},"update_available":null,"voltage":233}' with variables {'entity_id': 'select.wohnzimmer_tv_strom_indicator_mode', 'name': 'Indicator mode', 'this': <template TemplateStateFromEntityId(select.wohnzimmer_tv_strom_indicator_mode)>} and Template<template=({{ value_json.indicator_mode }}) renders=8>
2023-09-27 08:44:20.858 DEBUG (MainThread) [homeassistant.components.mqtt.models] Rendering incoming payload '{"child_lock":"LOCK","current":0.24,"energy":55.91,"indicator_mode":null,"linkquality":69,"power":13,"power_outage_memory":"on","state":"ON","update":{"installed_version":-1,"latest_version":-1,"state":null},"update_available":null,"voltage":233}' with variables {'entity_id': 'update.wohnzimmer_tv_strom', 'name': None, 'this': <template TemplateStateFromEntityId(update.wohnzimmer_tv_strom)>} and Template<template=({{ value_json['update']['installed_version'] }}) renders=8>
2023-09-27 08:44:20.858 DEBUG (MainThread) [homeassistant.components.mqtt.update] Non-dictionary JSON payload detected after processing payload '-1' on topic zigbee2mqtt/Wohnzimmer TV Strom
2023-09-27 08:44:20.859 DEBUG (MainThread) [homeassistant.components.mqtt.models] Rendering incoming payload '{"child_lock":"LOCK","current":0.24,"energy":55.91,"indicator_mode":null,"linkquality":69,"power":13,"power_outage_memory":"on","state":"ON","update":{"installed_version":-1,"latest_version":-1,"state":null},"update_available":null,"voltage":233}' with variables {'entity_id': 'update.wohnzimmer_tv_strom', 'name': None, 'this': <template TemplateStateFromEntityId(update.wohnzimmer_tv_strom)>} and Template<template=({{ value_json['update']['latest_version'] }}) renders=8>
2023-09-27 08:44:20.859 DEBUG (MainThread) [homeassistant.components.mqtt.models] Rendering incoming payload '{"child_lock":"LOCK","current":0.24,"energy":55.91,"indicator_mode":null,"linkquality":69,"power":13,"power_outage_memory":"on","state":"ON","update":{"installed_version":-1,"latest_version":-1,"state":null},"update_available":null,"voltage":233}' with variables {'entity_id': 'lock.wohnzimmer_tv_strom_child_lock', 'name': 'Child lock', 'this': <template TemplateStateFromEntityId(lock.wohnzimmer_tv_strom_child_lock)>} and Template<template=({{ value_json.child_lock }}) renders=8>
2023-09-27 08:44:20.906 DEBUG (MainThread) [huawei_solar.huawei_solar] Reading register 43006 with length 1 from slave 0
2023-09-27 08:44:20.907 DEBUG (MainThread) [pymodbus.logging] send: 0x0 0x1 0x0 0x0 0x0 0x6 0x0 0x3 0xa7 0xfe 0x0 0x1
2023-09-27 08:44:20.908 DEBUG (MainThread) [pymodbus.logging] Adding transaction 1
2023-09-27 08:44:20.921 DEBUG (MainThread) [pymodbus.logging] recv: 0x0 0x1 0x0 0x0 0x0 0x5 0x0 0x3 0x2 0x0 0x3c old_data:  addr=None
2023-09-27 08:44:20.921 DEBUG (MainThread) [pymodbus.logging] Processing: 0x0 0x1 0x0 0x0 0x0 0x5 0x0 0x3 0x2 0x0 0x3c
2023-09-27 08:44:20.921 DEBUG (MainThread) [pymodbus.logging] Factory Response[ReadHoldingRegistersResponse': 3]
2023-09-27 08:44:20.922 DEBUG (MainThread) [pymodbus.logging] Getting transaction 1
2023-09-27 08:44:20.924 ERROR (MainThread) [huawei_solar.huawei_solar] Aborting client creation due to error.
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/huawei_solar/huawei_solar.py", line 178, in create
    await huawei_solar._initialize()
  File "/usr/local/lib/python3.11/site-packages/huawei_solar/huawei_solar.py", line 107, in _initialize
    self.time_zone = (await self.get(rn.TIME_ZONE)).value
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/huawei_solar/huawei_solar.py", line 241, in get
    return (await self.get_multiple([name], slave))[0]
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/huawei_solar/huawei_solar.py", line 278, in get_multiple
    decoder = BinaryPayloadDecoder.fromRegisters(response.registers, byteorder=Endian.Big, wordorder=Endian.Big)
                                                                               ^^^^^^^^^^
  File "/usr/local/lib/python3.11/enum.py", line 784, in __getattr__
    raise AttributeError(name) from None
AttributeError: Big
2023-09-27 08:44:20.938 ERROR (MainThread) [homeassistant.config_entries] Error setting up entry SUN2000-8KTL-M1 01074300-002 for huawei_solar
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/huawei_solar/huawei_solar.py", line 178, in create
    await huawei_solar._initialize()
  File "/usr/local/lib/python3.11/site-packages/huawei_solar/huawei_solar.py", line 107, in _initialize
    self.time_zone = (await self.get(rn.TIME_ZONE)).value
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/huawei_solar/huawei_solar.py", line 241, in get
    return (await self.get_multiple([name], slave))[0]
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/huawei_solar/huawei_solar.py", line 278, in get_multiple
    decoder = BinaryPayloadDecoder.fromRegisters(response.registers, byteorder=Endian.Big, wordorder=Endian.Big)
                                                                               ^^^^^^^^^^
  File "/usr/local/lib/python3.11/enum.py", line 784, in __getattr__
    raise AttributeError(name) from None
AttributeError: Big

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 387, in async_setup
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/huawei_solar/__init__.py", line 201, in async_setup_entry
    raise err
  File "/config/custom_components/huawei_solar/__init__.py", line 88, in async_setup_entry
    primary_bridge = await HuaweiSolarBridge.create(
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/huawei_solar/bridge.py", line 66, in create
    client = await AsyncHuaweiSolar.create(host, port, slave_id)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/huawei_solar/huawei_solar.py", line 187, in create
    await client.close()
TypeError: object NoneType can't be used in 'await' expression
2023-09-27 08:44:20.956 DEBUG (MainThread) [homeassistant.components.websocket_api.http.connection] [547426789184] Boris from 192.168.2.21 (Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/118.0): Sending {"id":235,"type":"event","event":[{"type":"updated","entry":{"entry_id":"0fb10aad00efb384d23ddc00dbc8c83f","domain":"huawei_solar","title":"SUN2000-8KTL-M1\u000001074300-002","source":"user","state":"setup_error","supports_options":false,"supports_remove_device":false,"supports_unload":true,"pref_disable_new_entities":false,"pref_disable_polling":false,"disabled_by":null,"reason":null}}]}
2023-09-27 08:44:21.059 DEBUG (MainThread) [homeassistant.components.fritz.common] Using old hosts discovery method. (Mesh not supported or user option)
...
```

### Please confirm the following:

- [X] The problem is still present in the latest release of this integration.
- [X] I did not find an existing issue describing this problem.
- [X] I did upload the diagnostics-file that I could retrieve from  the 'Devices & Services Page'
- [X] I did upload the relevant debug logs (via 'Enable Debug Logging'-feature or by manually configuring HA logging)

**Comment by wlcrs** [comment]
Duplicate of #470 

TL;DR: upgrade to v1.2.7 of this integration.
